Widgets_CDAT/DictWidget/KeyValueEditor.py

```diff
@@ -0,0 +1,274 @@
+from PySide.QtCore import *
+from PySide.QtGui import *
+from functools import partial
+import pdb
+
+class InputChecker(QValidator):
+
+    inputInvalid = Signal()
+    correctInput = Signal()
+
+    def __init__(self, index):
+        super(InputChecker, self).__init__()
+        self.texts = []
+        self.index = index
+
+    def validate(self, input, pos):
+        if input in self.texts:
+            if self.texts.index(input) != self.index:
+                self.inputInvalid.emit()
+                return QValidator.Intermediate
+
+        self.correctInput.emit()
+        return QValidator.Acceptable
+
+    def setIndex(self, index):
+        self.index = index
+
+
+class DictEditor(QWidget):
+    dictEdited = Signal(dict)
+
+    def __init__(self):
+        super(DictEditor, self).__init__()
+        self.valid_keys = None
+        self.keys = []
+        self.values = []
+        self.textEdits = []
+        self.valueEdits = []
+        self.validators = []
+        self.row_count = 0
+        self.rows = QVBoxLayout()
+        self.button = QVBoxLayout()
+        self.wrap = QVBoxLayout()
+        add_button = QPushButton()
+        add_button.setText("New Line")
+        add_button.clicked.connect(self.insertRow)
+        self.clearing = False
+        self.button.addWidget(add_button)
+        self.setLayout(self.wrap)
+        self.wrap.addLayout(self.rows)
+        self.wrap.addLayout(self.button)
+
+    def colorInvalid(self, line_edit):
+        line_edit.setStyleSheet("color: rgb(255, 0, 0);")
+
+    def colorValid(self, line_edit):
+        line_edit.setStyleSheet("color: rgb(0, 0, 0);")
+
+
+    # Update Combo Boxes
+    def updateCBoxes(self, cur_box):
+        cur_text = cur_box.itemText(cur_box.currentIndex())
+        selectedItems = []
+        for box in self.cBoxes:
+            check_text = box.itemText(box.currentIndex())
+
+            selectedItems.append(check_text)
+
+            # check if boxes match and set to blank if so
+            if box != cur_box:
+                if check_text == cur_text:
+                    box.setCurrentIndex(0)
+
+        return selectedItems
+
+
+    # updates dict based on lineedit values
+    def updateEditTexts(self):
+        texts = []
+        for item in self.textEdits:
+            texts.append(item.text())
+        for i in self.validators:
+            i.texts = self.keys
+
+
+    # update lists and selectedItems
+    def updateLists(self, left, right, index=None):
```

Comment:
What use is the `index` keyword argument here? You set `index` in both the `if` and the `else`.